pkg/controller/cassandradatacenter/controller/ports.go

```diff
@@ -0,0 +1,18 @@
+package controller
```

Comment:
This currently isn't used, but included for discussion.

Following the discussion on constants, it was determined that any value that's used in more than one place probably should become a constant. The port numbers (and their names) are used in both the container and service definitions, so IMO they should be constants.

Issue is, services hold `ServicePort` structs and containers hold `ContainerPort` structs.

Pending questions:
- do the `ServicePort` and `ContainerPort` names need to match?
- what if we need to define service or container ports that leverage other fields of their respective structs?

Reply:
I somehow came up with this ...

```
package controller

import v1 "k8s.io/api/core/v1"

type port struct {
	name string
	port int32
}

type PortFactory struct {
	servicePort   v1.ServicePort
	containerPort v1.ContainerPort
}

var cassandraJmxPort = port{"jmx", 7199}

func (myPort *PortFactory) ContainerPort(p port) v1.ContainerPort {
	myPort.containerPort = v1.ContainerPort{Name: p.name, ContainerPort: p.port}

	// other operations on myPort.containerPort

	return myPort.containerPort
}

func (myPort *PortFactory) ServicePort(p port) v1.ServicePort {

	myPort.servicePort = v1.ServicePort{Name: p.name, Port: p.port}

	// other operations on myPort.servicePort

	return myPort.servicePort
}

/*
	Would be used as

	ctx := portContext{
		name: "",
		port: 1234,
	}

	var servicePort v1.ServicePort = PortFactory.ServicePort(PortFactory{}, ctx)
*/
```

Reply:
Maybe this is even better:

```
package controller

import v1 "k8s.io/api/core/v1"

type port struct {
	name string
	port int32
}

type PortFactory interface {
	ContainerPort() v1.ContainerPort
	ServicePort() v1.ServicePort
}

var cassandraJmxPort = port{"jmx", 7199}

func (myPort port) ContainerPort() v1.ContainerPort {

	containerPort := v1.ContainerPort{Name: myPort.name, ContainerPort: myPort.port}

	// other operations

	return containerPort
}

func (myPort port) ServicePort() v1.ServicePort {

	servicePort := v1.ServicePort{Name: myPort.name, Port: myPort.port}

	// other operations
	
	return servicePort
}
```

port struct itself would  have these methods on it, from which  you can form whatever specific port you want

Reply:
In this case, there's nothing in `PortFactory` that we really need to create the results. In many cases, we'd simply use functions `func ContainerPort()...` and `func ServicePort` to do the same. The receiver in this case doesn't really bring any value. Also, we can do it without the `port` type, just provide `name, port` vars into the creator functions.

We do a similar thing in gocql when creating a new Session object via the `NewSession(options...)` call.

Reply:
ah yes, we can omit that interface completely, thats true ... I like that from usage point of view, there is some "generic port" you call functions on to "morph" into whatever you want